src/server.rs

```diff
@@ -197,7 +194,7 @@ impl Default for LspServerBuilder {
 
 #[allow(dead_code)]
 pub struct LspServer {
-    client: Arc<Mutex<Option<Client>>>,
+    client: Option<Client>,
```

Comment:
This was so that we could have _interior mutability_ on the client as we needed to explicitly drop it when the server is told to shutdown. As dropping the client is the only way to make it close its channel and signal to the receiving code that there are not more messages coming. 

With the change to be able to explicitly close the client we no longer need to drop it and can let it be dropped when the server itself is dropped.